sklearn/manifold/tests/test_t_sne.py

```diff
@@ -542,3 +542,32 @@ def test_index_offset():
     # Make sure translating between 1D and N-D indices are preserved
     assert_equal(_barnes_hut_tsne.test_index2offset(), 1)
     assert_equal(_barnes_hut_tsne.test_index_offset(), 1)
+
+
+def test_accessible_kl_divergence():
+    # Ensures that the accessible kl_divergence matches the computed value
+    random_state = check_random_state(0)
+    X = random_state.randn(100, 2)
+    tsne = TSNE(n_iter_without_progress=2, verbose=2,
+                random_state=0, method='exact')
+
+    old_stdout = sys.stdout
+    sys.stdout = StringIO()
+    try:
+        tsne.fit_transform(X)
+    finally:
+        out = sys.stdout.getvalue()
+        sys.stdout.close()
+        sys.stdout = old_stdout
+
+    # The output needs to contain the accessible kl_divergence as the error at
+    # the last iteration
+    lines_out = out.split('\n')[::-1]
+    for line in lines_out:
+        if 'Iteration' in line:
+            start_pos = line.find('error = ')
```

Comment:
This could be written a bit more neatly with regular expressions or:
```
_, _, error = line.partition('error = ')
if error:
    error, _, _ = error.partition(',')
```
or some such. I find the current code very hard to read, given its mix of find, slicing, split and replace.